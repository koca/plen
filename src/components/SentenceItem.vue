<template>
  <div class="sentence">
    <div class="mb-4 p-4 relative shadow bg-white">
      <span class="absolute pin-r pin-t text-grey p-4">{{id}}</span>
      <span v-text="sentence.en || sentence.obj.en" highlight="hi" />
      <div class="my-2 border-b-2 w-4"></div>
      <span :bold="true" v-text="sentence.pl || sentence.obj.pl" />
    </div>
  </div>
</template>

<script>
export default {
  props: {
    sentence: {
      type: [Object, Array],
      required: true
    },
    id: {
      type: [String, Number],
      required: true
    }
  }
};
</script>

<style>
</style>
