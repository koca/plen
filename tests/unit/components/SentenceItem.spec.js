import { shallowMount } from "@vue/test-utils";
import SentenceItem from "@/components/SentenceItem.vue";

describe("SentenceItem.vue", () => {
  let wrapper;
  const sentence = { en: "en1", pl: "pl1" };

  beforeEach(() => {
    wrapper = shallowMount(SentenceItem, {
      propsData: {
        sentence,
        id: 0
      }
    });
  });
  it("renders properly", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
  it("displays a sentence", () => {
    expect(wrapper.find(".sentence").text()).toContain("en1");
    expect(wrapper.find(".sentence").text()).toContain("pl1");
  });
});
