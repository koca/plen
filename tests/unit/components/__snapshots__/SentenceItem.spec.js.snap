// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SentenceItem.vue renders properly 1`] = `
<div class="sentence">
  <div class="mb-4 p-4 relative shadow bg-white"><span class="absolute pin-r pin-t text-grey p-4">0</span> <span highlight="hi">en1</span>
    <div class="my-2 border-b-2 w-4"></div> <span bold="true">pl1</span></div>
</div>
`;
